package.json

```diff
@@ -78,6 +90,7 @@
     "ember-cli-release": "^0.2.9",
     "ember-cli-shims": "^1.2.0",
     "ember-cli-sri": "^2.1.0",
+    "ember-cli-typescript": "^1.1.6",
```

Comment:
I believe npm would error on this (same name as package) ?

Reply:
I haven't checked on publishing, but certainly none of the local operations complain. You'd never be able to publish _any_ package which wanted to use itself otherwise.

Reply:
Yes, for example typescript is written in typescript and thus has itself as a dependency. Inception!

Reply:
@dwickern not really, they do not add it in package.json, I believe they use a last known good version here  https://github.com/Microsoft/TypeScript/tree/master/lib 

And yes that's what I meant @chriskrycho, it won't publish if same name is inside dependencies.

Reply:
It's there:
https://github.com/Microsoft/TypeScript/blob/master/package.json#L83

Reply:
Okay this will not cause any problem. Releasing will work too as it is in devDependencies. I just checked with a dummy package :) 

Sorry for troubling :D